universal-application-tool-0.0.1/app/repository/VersionRepository.java

```diff
@@ -0,0 +1,328 @@
+package repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import io.ebean.Ebean;
+import io.ebean.EbeanServer;
+import io.ebean.Transaction;
+import java.util.Optional;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+import models.LifecycleStage;
+import models.Program;
+import models.Question;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.db.ebean.EbeanConfig;
+import services.program.BlockDefinition;
+import services.program.ProgramDefinition;
+import services.program.ProgramQuestionDefinition;
+
+public class VersionRepository {
+
+  private final EbeanServer ebeanServer;
+  private final ProgramRepository programRepository;
+  private final Logger LOG = LoggerFactory.getLogger(VersionRepository.class);
+
+  @Inject
+  public VersionRepository(EbeanConfig ebeanConfig, ProgramRepository programRepository) {
+    this.ebeanServer = Ebean.getServer(checkNotNull(ebeanConfig).defaultServer());
+    this.programRepository = checkNotNull(programRepository);
+  }
+
+  public void publishNewSynchronizedVersion() {
```

Comment:
suggestion (blocking): document this public method

Reply:
if there are no DRAFT programs, this method would continue copying all programs. That should be clear in the doc string.

Reply:
Done.